langchain/callbacks/aim_callback.py

```diff
@@ -0,0 +1,602 @@
+            if self._run_hash:
+                self._run = Run(
+                    self._run_hash,
+                    repo=self.repo,
+                    system_tracking_interval=self.system_tracking_interval,
+                )
+            else:
+                self._run = Run(
+                    repo=self.repo,
+                    experiment=self.experiment_name,
+                    system_tracking_interval=self.system_tracking_interval,
+                    log_system_params=self.log_system_params,
+                )
+                self._run_hash = self._run.hash
+
+        if args:
+            for key, value in args.items():
+                self._run.set(key, value, strict=False)
+
+    @classmethod
+    def _flatten_dict(
+        cls, nested_dict: Dict[str, Any], parent_key: str = "", sep: str = "_"
+    ) -> Iterable[Tuple[str, Any]]:
+        """
+        Generator that yields flattened items from a nested dictionary for a flat dict.
+
+        Parameters:
+            nested_dict (dict): The nested dictionary to flatten.
+            parent_key (str): The prefix to prepend to the keys of the flattened dict.
+            sep (str): The separator to use between the parent key and the key of the
+                flattened dictionary.
+
+        Yields:
+            (str, any): A key-value pair from the flattened dictionary.
+        """
+        for key, value in nested_dict.items():
+            new_key = parent_key + sep + key if parent_key else key
+            if isinstance(value, dict):
+                yield from AimCallbackHandler._flatten_dict(value, new_key, sep)
+            else:
+                yield new_key, value
+
+    @classmethod
+    def flatten_dict(
+        cls, nested_dict: Dict[str, Any], parent_key: str = "", sep: str = "_"
+    ) -> Dict[str, Any]:
+        """Flattens a nested dictionary into a flat dictionary.
+
+        Parameters:
+            nested_dict (dict): The nested dictionary to flatten.
+            parent_key (str): The prefix to prepend to the keys of the flattened dict.
+            sep (str): The separator to use between the parent key and the key of the
+                flattened dictionary.
+
+        Returns:
+            (dict): A flat dictionary.
+
+        """
+        flat_dict = {
+            k: v
+            for k, v in AimCallbackHandler._flatten_dict(nested_dict, parent_key, sep)
+        }
+        return flat_dict
+
+    @classmethod
+    def get_text_stats(
+        cls,
+        text: str,
+        complexity_metrics: bool = True,
+    ) -> dict:
+        """Analyze text using textstat.
+
+        Parameters:
+            text (str): The text to analyze.
+            complexity_metrics (bool): Whether to compute complexity metrics.
+
+        Returns:
+            (dict): A dictionary containing the complexity metrics.
+        """
+        try:
+            import textstat  # noqa: F401
+        except ImportError:
+            raise ImportError(
```

Comment:
I think we can just skip text stats if the textstat package is not installed, thoughts?

Reply:
sounds good, but the complexity_metrics flag will not function properly, i.e. switching it on and off will not make any difference on the processes.